File: iosApp/iosApp/ConfigurationDetail/Views/ConfigurationDetailState.swift
//
// Created by Julius Canute on 11/4/20.
//

import Foundation

enum ConfigurationDetailState {
    case initialState
    case loadEnvironmentConfiguration(LoadEnvironmentConfigurationState)
    case showChoiceConfiguration(ShowChoiceConfigurationState)
    case showEditableState(ShowEditableState)
}

struct LoadEnvironmentConfigurationState {
    let items: [ItemState]
}

struct ShowChoiceConfigurationState {
    let description: String
    let items: [ChoiceItem]
    let currentSelection: Int
    let key: String
}

struct ShowEditableState {
    let description: String
    let value: String
    let key: String
}

File: app/src/main/java/com/juliuscanute/multiconfig/ui/configdetail/ConfigurationDetailState.kt
package com.juliuscanute.multiconfig.ui.configdetail

import com.juliuscanute.multiconfig.ui.adapter.ItemState
import com.juliuscanute.multiconfig.model.Item

sealed class ConfigurationDetailState {
    data class LoadEnvironmentConfigurationState(val items: List<ItemState>) : ConfigurationDetailState()
    data class ShowChoiceConfigurationState(
        val description: String,
        val items: ArrayList<Item>,
        val currentSelection: Int,
        val key: String
    ) :
        ConfigurationDetailState()

    data class ShowEditableState(
        val description: String, val value: String, val key: String
    ) : ConfigurationDetailState()
}